atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -167,21 +171,44 @@ def assemble_srpm_url(self, base_url, srpm_filename, sign_key=None):
             srpm_path = self.pathinfo.rpm(srpm_info)
         return '/'.join([base_url, srpm_path])
 
-    def get_remote_urls(self):
-        """Fetch remote source urls
+    def _get_remote_urls_helper(self, koji_build):
+        """Fetch remote source urls from specific build
 
+        :param koji_build: dict, koji build
         :return: str, URL pointing to remote sources
         """
-        self.log.debug('get remote_urls: %s', self.koji_build_id)
-        archives = self.session.listArchives(self.koji_build_id, type='remote-sources')
+        self.log.debug('get remote_urls: %s', koji_build['build_id'])
+        archives = self.session.listArchives(koji_build['build_id'], type='remote-sources')
         self.log.debug('archives: %s', archives)
-        remote_sources_path = self.pathinfo.typedir(self.koji_build, btype='remote-sources')
+        remote_sources_path = self.pathinfo.typedir(koji_build, btype='remote-sources')
         remote_sources_urls = []
 
         for archive in archives:
             if archive['type_name'] == 'tar':
-                remote_source = os.path.join(remote_sources_path, archive['filename'])
+                remote_source = {}
+                remote_source['url'] = os.path.join(remote_sources_path, archive['filename'])
+                remote_source['dest'] = '-'.join([koji_build['nvr'], archive['filename']])
                 remote_sources_urls.append(remote_source)
+        return remote_sources_urls
+
+    def get_remote_urls(self):
+        """Fetch remote source urls from all builds
+
+        :return: list, dicts with URL pointing to remote sources
+        """
+        remote_sources_urls = []
+
+        remote_sources_urls.extend(self._get_remote_urls_helper(self.koji_build))
+
+        koji_build = self.koji_build
+
+        while True:
```

Comment:
Maybe
`while 'parent_build_id' in koji_build['extra']['image']:`
to make it more readable?

Reply:
ok